tests/integration/live_data/zookeeper_tests.py

```diff
@@ -48,3 +53,37 @@ def test_create_client_with_credentials(self):
 
         secrets.get_simple.assert_called_with("secret/zk-user")
         self.assertEqual(list(client.auth_data), [("digest", "myzkuser:hunter2")])
+
+    def test_create_client_uses_correct_handler(self):
+        secrets = mock.Mock(spec=SecretsStore)
+        client = zookeeper_client_from_config(
+            secrets, {"zookeeper.hosts": "%s:%d" % zookeeper_endpoint.address}
+        )
+        assert isinstance(client.handler, SequentialThreadingHandler)
+
+
+class ZooKeeperGeventTests(unittest.TestCase):
+    def setUp(self):
+        # We patch `socket` just to make sure that the gevent handler is chosen.
+        # Nothing is special about `socket` in particular.
```

Comment:
If we don't care about socket in particular, I would suggest patching the same way baseplate does, aka patch_all + baseplate.monkey.patch_queue

Reply:
We can't do this because some of what `patch_all` patches we can't "unpatch" (as far as I know). For example, we have no way to unpatch `time.sleep` as it is a builtin and `importlib.reload` doesn't unload it.

Reply:
(I'll add a comment to this effect)

Reply:
It might make sense to review how we handle gevent tests in a future PR as the current state of things is very fiddly.

Reply:
https://docs.python.org/3/library/importlib.html#importlib.reload

You might just have to import `time` before patching